### ISSUE TITLE
GET api/v1/users/{username}/feed 500 errors

### ISSUE THREAD

**Issue by nstratos** [hummingbird-me/hummingbird#539]
Hello,

Getting a user's feed seems to be giving random(?) 500 errors. Some examples:

| Username          |                              HTTP Request                                                 |    Result      |
| --------------------- | ------------------------------------------------------------------------------------ | --------------- |
| kitataka              | GET https://hummingbird.me/api/v1/users/kitataka/feed              | SUCCESS  |
| cybrox                | GET https://hummingbird.me/api/v1/users/cybrox/feed                | 500             |
| thekitsune          | GET https://hummingbird.me/api/v1/users/thekitsune/feed          | SUCCESS  |
| glassarmour       | GET https://hummingbird.me/api/v1/users/glassarmour/feed       | 500             |
| toomanydangos | GET https://hummingbird.me/api/v1/users/toomanydangos/feed | 500             |
| auxoran             | GET https://hummingbird.me/api/v1/users/auxoran/feed              | 500             |
| Meiwa                | GET https://hummingbird.me/api/v1/users/Meiwa/feed                |  SUCCESS |

Hummingbird forums [thread](http://forums.hummingbird.me/t/get-api-v1-users-username-feed-500-errors).

**Comment by vevix** [comment]
Ugh this sucks. I can't look into this right now but i'll get to it ASAP if @NuckChorris doesn't first.

**Comment by nstratos** [comment]
After successfully running Hummingbird locally (all dependencies and database structure had no errors), I created an admin user and imported a few anime from MAL. When I tried to access the user's feed I got: `Association named 'genres' was not found on User`
Line 313 on file hummingbird/app/api/api_v1.rb
```ruby
stories = user.stories.for_user(current_user).order('updated_at DESC').includes(:substories, :user, :target, target: :genres).page(params[:page]).per(20)
```
The other API endpoints that I tried worked fine.

Did I miss any extra steps or do you think this might be causing the errors?

**Comment by vevix** [comment]
Yep that will be the issue introduced in #533

**Comment by NuckChorris** [comment]
Yeah it would, I forgot stories can also target users.

**Comment by DNA** [comment]
My bad, I didn't added all the possible stories on my feed when testing this, so I didn't got this error before submiting the PR. Unfortunately nesting eager loading of polymophic association like I did will be available only in rails 5 :disappointed: https://github.com/rails/rails/pull/17479

But I got and workaround for it. I'm sending a new PR to see if it's okay. (some help with tests to prevent errors like this in the future would be appreciated :smile: )
